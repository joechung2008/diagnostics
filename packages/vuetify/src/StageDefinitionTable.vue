<script lang="ts" setup>
import { computed } from 'vue'
import type { StageDefinitionProps } from './StageDefinitionTable.types'

const props = defineProps<StageDefinitionProps>()

const headers = [
  { title: 'Key', value: 'key' },
  { title: 'Value', value: 'value' },
]

const items = computed(() =>
  Object.entries(props.stageDefinition).map(([key, value]) => ({
    key,
    value: value.join(', '),
  }))
)
</script>

<template>
  <h2 class="text-h4">Stage Definitions</h2>
  <v-data-table :headers="headers" :items="items" hide-default-footer />
</template>
