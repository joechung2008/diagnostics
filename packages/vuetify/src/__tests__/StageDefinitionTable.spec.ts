import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import StageDefinitionTable from '../StageDefinitionTable.vue'

describe('StageDefinitionTable', () => {
  it('renders stage definitions', () => {
    const stageDefinition = {
      build: ['compile', 'link'],
      test: ['unit', 'integration'],
    }
    const wrapper = mount(StageDefinitionTable, {
      props: { stageDefinition },
    })
    expect(wrapper.exists()).toBe(true)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
