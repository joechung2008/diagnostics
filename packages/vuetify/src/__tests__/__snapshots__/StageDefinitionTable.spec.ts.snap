// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StageDefinitionTable > renders stage definitions 1`] = `
"<h2 class="text-h4">Stage Definitions</h2>
<div class="v-table v-table--has-top v-table--has-bottom v-theme--light v-table--density-default v-data-table">
  <!---->
  <div class="v-table__wrapper">
    <table>
      <!---->
      <thead>
        <tr>
          <th class="v-data-table__td v-data-table-column--align-start v-data-table__th" colspan="1" rowspan="1">
            <div class="v-data-table-header__content"><span>Key</span>
              <!---->
              <!---->
            </div>
          </th>
          <th class="v-data-table__td v-data-table-column--align-start v-data-table__th" colspan="1" rowspan="1">
            <div class="v-data-table-header__content"><span>Value</span>
              <!---->
              <!---->
            </div>
          </th>
        </tr>
        <!---->
      </thead>
      <!---->
      <tbody>
        <!---->
        <tr class="v-data-table__tr">
          <td class="v-data-table__td v-data-table-column--align-start">build</td>
          <td class="v-data-table__td v-data-table-column--align-start">compile, link</td>
        </tr>
        <!---->
        <tr class="v-data-table__tr">
          <td class="v-data-table__td v-data-table-column--align-start">test</td>
          <td class="v-data-table__td v-data-table-column--align-start">unit, integration</td>
        </tr>
        <!---->
        <!---->
      </tbody>
      <!---->
      <!---->
    </table>
  </div>
  <!---->
</div>"
`;
